ReplayMessenger.device.lib.nut

```diff
@@ -93,6 +109,51 @@ class ReplayMessenger {
         return _mm.send(messageName, data, null, _retryInterval, metadata);
     }
 
+    function beforeReplay(loggerName, callback = null) {
+      if (loggerName == null && callback == null) { //assume the user wants to de-register the callback for the "default" logger
+        loggerName = "default";
+      }
+
+      if (typeof loggerName == "string" && callback == null) { //de-register callback
+        if (loggerName in _beforeReplay) delete _beforeReplay[loggerName]
+        return;
+      }
+
+      if (typeof loggerName == "function" && callback == null) { //assume this is the callback provided for "default" logger
+        callback = loggerName;
+        loggerName = "default";
+      }
+
+      _beforeReplay[loggerName] <- callback;
+    }
+
+    function afterReplay(loggerName, callback = null) {
+      if (loggerName == null && callback == null) { //assume the user wants to de-register the callback for the "default" logger
+        loggerName = "default";
+      }
+
+      if (typeof loggerName == "string" && callback == null) { //de-register callback
+        if (loggerName in _afterReplay) delete _afterReplay[loggerName]
+        return;
+      }
+
+      if (typeof loggerName == "function" && callback == null) { //assume this is the callback provided for "default" logger
+        callback = loggerName;
+        loggerName = "default";
+      }
+
+      _afterReplay[loggerName] <- callback;
+    }
+
+    function setLoggerReadOptions(loggerName, options=null) {
+      if (typeof loggerName == "table" && options == null) { //assume these are the read options provided for the "default" logger
```

Comment:
This is a vicious practice to dynamically switch positions of the arguments! Let's please not do so!

a) it is very confusion and
b) setLoggerReadOptions(null, {...}); would make it fail.

If you want to have named optional parameters let's just use a map with two values. But I don't really think it's worth it.